code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm

```diff
@@ -136,12 +136,22 @@
 	var/vine_grab_distance = 5
 	/// Whether or not this plant is ghost possessable
 	var/playable_plant = TRUE
-
 	discovery_points = 2000
 
+/mob/living/simple_animal/hostile/venus_human_trap/Initialize()
+	remove_verb(/mob/living/verb/pulled) //No pulling people into the vines
+	. = ..()
+
 /mob/living/simple_animal/hostile/venus_human_trap/Life()
 	. = ..()
 	pull_vines()
+	var/turf/our_turf = get_turf(src)
+	for(var/obj/check_vine in our_turf.contents)
+		if(istype(check_vine, /obj/structure/spacevine))//Heal if we are on vines
+			adjustHealth(-maxHealth*0.05)
```

Comment:
You should use the `/obj/effect/temp_visual/heal` to provide a visual indicator that they are being healed when not at max health

Reply:
Can't a new overlay subtype be created with a higher layer?

Reply:
Probably, again I'm not available at the moment sadly.